frontend/src/metabase/common/hooks/use-confirmation.tsx

```diff
@@ -10,6 +11,7 @@ export type ConfirmationState = {
   onCancel?: () => void;
   confirmButtonText?: string;
   cancelButtonText?: string;
+  confirmButtonProps?: Omit<ButtonProps, "onClick" | "children">;
```

Comment:
same type as the `ConfirmModal` prop

https://github.com/metabase/metabase/blob/afded810c1b46b53b1ebf2c3c77875f0b3c6cc44/frontend/src/metabase/common/components/ConfirmModal/ConfirmModal.tsx#L20